rasa/core/policies/ted_policy.py

```diff
@@ -347,22 +355,32 @@ def __init__(
         self.only_e2e = TEXT in self.fake_features and INTENT not in self.fake_features
 
         self._label_data: Optional[RasaModelData] = None
-        self.data_example: Optional[Dict[Text, List[np.ndarray]]] = None
+        self.data_example: Optional[Dict[Text, Dict[Text, List[FeatureArray]]]] = None
 
         self.tmp_checkpoint_dir = None
         if self.config[CHECKPOINT_MODEL]:
             self.tmp_checkpoint_dir = Path(rasa.utils.io.create_temporary_directory())
 
+    @staticmethod
+    def model_class() -> Type[RasaModel]:
+        return TED
```

Comment:
Curious why this method is needed. Is it gonna be used dynamically later? Now it simply returns a constant and this constant doesn't seem to be used somewhere else except by this particular class.

Reply:
Yes, inside `IntentTEDPolicy`